wrapper/src/main/java/software/amazon/jdbc/RoundRobinHostSelector.java

```diff
@@ -107,27 +107,24 @@ private void createCacheEntryForHosts(
       final @NonNull List<HostSpec> hosts,
       final @Nullable Properties props)
       throws SQLException {
-    final List<HostSpec> hostsMissingCacheEntry = new ArrayList<>();
     final List<HostSpec> hostsWithCacheEntry = new ArrayList<>();
     for (final HostSpec host : hosts) {
       if (roundRobinCache.get(host.getHost()) != null) {
         hostsWithCacheEntry.add(host);
-      } else {
-        hostsMissingCacheEntry.add(host);
       }
     }
 
-    if ((hostsMissingCacheEntry.isEmpty() && !hostsWithCacheEntry.isEmpty())) {
+    if (!hostsWithCacheEntry.isEmpty()) {
       for (final HostSpec host : hosts) {
         roundRobinCache.put(
             host.getHost(),
             roundRobinCache.get(hostsWithCacheEntry.get(0).getHost()),
             DEFAULT_ROUND_ROBIN_CACHE_EXPIRE_NANO);
       }
-    } else if (hostsWithCacheEntry.isEmpty()) {
+    } else {
       final RoundRobinClusterInfo roundRobinClusterInfo = new RoundRobinClusterInfo();
       updateCachePropertiesForRoundRobinClusterInfo(roundRobinClusterInfo, props);
-      for (final HostSpec host : hostsMissingCacheEntry) {
+      for (final HostSpec host : hosts) {
```

Comment:
does that mean that a new cache entry will be added for every host? including those that already had a cache entry?

let's add comments explaining each part of the conditions

Reply:
Yes, each host will have a cache entry. If the host already had an entry, it will just be updated.